### PR TITLE
mitosheet: handle taskpane crashes locally

### PR DESCRIPTION
**Pull request by naterush**
# Description

Handles taskpane crashing issues locally, allowing the users to just reopen taskpane and avoid crashing the entire sheet.

**Before merging, make sure to clean up the Values Tab.**

# Testing

Open the values tab. Check resulting logs.

**Before merging, make sure to clean up the Values Tab.**

# Documentation

No.

### CODE REVIEW

File: mitosheet/src/mito/components/taskpanes/DefaultTaskpane/DefaultTaskpaneErrorBoundry.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+            return {
+                ...prevUIState,
+                currOpenTaskpane: {type: TaskpaneType.NONE}
+            }
+        })
+
+        void this.props.mitoAPI?.log('taskpane_render_failed', {
+            js_error: error,
+            js_error_info: errorInfo,
+            js_taskpane_header: this.props.taskpaneHeader
```
Comment (aarondr77) [inline mitosheet/src/mito/components/taskpanes/DefaultTaskpane/DefaultTaskpaneErrorBoundry.tsx:39]: This often contains the column header that we would not want to log

Comment (aarondr77) [reply mitosheet/src/mito/components/taskpanes/DefaultTaskpane/DefaultTaskpaneErrorBoundry.tsx:39]: Instead, what if we log the taskpane type. Probably more useful anyways

**Review by aarondr77 — COMMENTED** [review]
LGTM after tiny cleanup + is it possible to add a frontend test for this? I can't exactly think how unless we did some inject code into the taskpane that errors??

File: mitosheet/src/mito/components/taskpanes/DefaultTaskpane/DefaultTaskpaneErrorBoundry.tsx (hunk 1 above)
Comment (naterush) [reply mitosheet/src/mito/components/taskpanes/DefaultTaskpane/DefaultTaskpaneErrorBoundry.tsx:39]: Really hard to get honestly, and we don't have the taskpane type in every case. Rather, I'm going to just remove this for now -- and we'll see if we need to add it back.

**Pull request merged**